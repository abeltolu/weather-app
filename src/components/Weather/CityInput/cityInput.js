import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import {
    InputContainer,
    Input,
    Button
} from '../../App/common';

export default class CityInput extends React.Component {

    constructor(){
        super();
    }

    componentDidMount(){
    }

    render() {
        const { city, handleCityChange } = this.props;
    
        return (
            <Fragment>
                <InputContainer>

                    <Input 
                        name="city" 
                        placeholder="Enter your City"
                        ref={input => {
                            this.cityInput = input;
                            return this.cityInput;
                        }} />

                    <Button onClick={(e) => {
                        const city = this.cityInput.value;
                        city.length > 0 && handleCityChange(city);
                    }}>Search</Button>
                </InputContainer>
            </Fragment>
        );
    }
}

CityInput.propTypes = {
    city: PropTypes.string,
    handleCityChange: PropTypes.func.isRequired,
}