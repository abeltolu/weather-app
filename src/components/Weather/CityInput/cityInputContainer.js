import { connect } from 'react-redux';
import CityInput from './cityInput';

//import the actions that need to be dispatched here
import { fetchData } from '../../../actions/weather';

function mapStateToProps(state){
    return {
        city: state.weather.city
    };
}

function mapDispatchToProps(dispatch){
    return {
        handleCityChange: (city) => {
            dispatch(fetchData(city));
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(CityInput);