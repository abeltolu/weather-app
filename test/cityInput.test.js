import React from "react";
import renderer from 'react-test-renderer';
import { render, mount, shallow } from "enzyme";
import CityInput from "../src/components/Weather/CityInput/cityInput";

describe("<CityInput />", () => {
    
    it("should render snapshot", () => {
        const wrapper = shallow(<CityInput handleCityChange={e => console.log(e)} />);
        expect(wrapper).toMatchSnapshot();
    });
});