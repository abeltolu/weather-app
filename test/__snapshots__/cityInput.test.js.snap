// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityInput /> should render snapshot 1`] = `
<Fragment>
  <styled.div>
    <styled.input
      name="city"
      placeholder="Enter your City"
    />
    <styled.button
      onClick={[Function]}
    >
      Search
    </styled.button>
  </styled.div>
</Fragment>
`;
